testsuite/support.py

```diff
@@ -83,6 +83,25 @@ def print_results(self):
         print("Test failed." if self.total_errors else "Test passed.")
 
 
+class InMemoryReport(BaseReport):
+    """
+    Collect the results in memory, without printing anything.
+    """
+
+    def __init__(self, options):
+        BaseReport.__init__(self, options)
```

Comment:
Why not use `super`? python 2/3 compatability?

Reply:
If BaseReport is not a new-style class I think you need to do this. I don't recall what kind of class it is, though

Reply:
I will do super :)  BaseReport is new style (class BaseReport(object):)

I just remembered that pycodestuly runs with a log of python versions...and while working on this, I was lazy and just went with this syntax. Will change to super.

Thanks:)

Reply:
I think 2.6+ supports new-style classes + super so you're safe :)

Reply:
Will use super :)

@sigmavirus24 , can you please confirm whether the new tests are the type of tests that you would like to see for this change?

If you are happy with this style of testing, I can continue to write more test so that they will touch all braches.

Thanks!

Reply:
Yeah, these tests look fantastic, @adiroiban :)